interface/gcp.py

```diff
@@ -17,54 +20,110 @@ def __init__(self, drive_client: Resource, subject=None):
         self.drive = drive_client
         self.subject = subject
 
-    def set_drive_permissions(self,
-                              team_name: str,
-                              drive_id: str,
-                              emails: List[str],
-                              delete_permissions=DELETE_OLD_DRIVE_PERMISSIONS):
+    def get_drive_parents(self, drive_id: str) -> List[str]:
+        """
+        Retrieves list of parents of the given Drive folder, returned as ID
+        strings.
+        """
+        # pylint: disable=no-member
+        file = self.drive.files()\
+            .get(fileId=drive_id, fields="parents")\
+            .execute()
+        if 'parents' in file:
+            parents: List[str] = file['parents']
+            if len(parents) > 0:
+                return parents
+        return []
+
+    def get_drive_permissions(self, drive_id: str) -> List[GCPDrivePermission]:
+        """
+        Retrieves list of permissions present on the given Drive item. It
+        pages through all results and returns a subset of permission fields,
+        as defined in `GCPPermission`.
+        """
+        perms: List[GCPDrivePermission] = []
+        page_count = 0
+        fields = [
+            "permissions/id",
+            "permissions/emailAddress",
+        ]
+        # See http://googleapis.github.io/google-api-python-client/docs/dyn/drive_v3.permissions.html#list # noqa
+        # pylint: disable=no-member
+        next_page_req = self.drive.permissions()\
+            .list(fileId=drive_id,
+                  fields=", ".join(fields),
+                  pageSize=50)
+        while next_page_req is not None:
+            list_res = next_page_req.execute()
+            permissions: List[Any] = list_res['permissions']
+            for p in permissions:
+                if 'emailAddress' in p:
+                    perm = GCPDrivePermission(p['id'], p['emailAddress'])
+                    perms.append(perm)
```

Comment:
Abstract pagination away with Python generators?

Reply:
Eh took a look at it and seems unnecessary and convoluted? Happy to take a suggestion if you believe theres a way to make this more succinct by setting up a generator for this single request (none of the others used, including parent listing, requires paging)

Reply:
Oh. It just makes it easier for me to read. 

```python
def get_paginated_permissions(self, drive_id: str, fields: List[str]): Iterator[Any]:
	req = self.drive.permissions()\
		.list(fileId=drive_id,
			  fields=', '.join(fields),
			  pageSize=50)
	while req is not None:
		resp = req.execute()
		for perm in resp['permissions']:
			yield perm
		req = self.drive.permissions().list_next(req, resp)
	raise StopIteration
```

Reply:
I tried this, but then the tests started failing. Help.

```diff
diff --git a/interface/gcp.py b/interface/gcp.py
index 78830b5..e6a29da 100644
--- a/interface/gcp.py
+++ b/interface/gcp.py
@@ -1,5 +1,5 @@
 """Utility classes for interacting with Google APIs"""
-from typing import Any, List
+from typing import Any, List, Iterator
 from googleapiclient.discovery import Resource
 import logging
 
@@ -35,6 +35,27 @@ class GCPInterface:
                 return parents
         return []
 
+    def get_paginated_permissions(self,
+                                  drive_id: str,
+                                  fields: List[str]) -> Iterator[Any]:
+        """
+        Retrieve generator of permissions present on the given Drive item.
+
+        Pages through them so you don't have to.
+        """
+        req = self.drive.permissions()\
+            .list(fileId=drive_id,
+                  fields=', '.join(fields),
+                  pageSize=50)
+
+        while req is not None:
+            resp = req.execute()
+            for perm in resp['permissions']:
+                yield perm
+            req = self.drive.permissions().list_next(req, resp)
+
+        raise StopIteration
+
     def get_drive_permissions(self, drive_id: str) -> List[GCPDrivePermission]:
         """
         Retrieves list of permissions present on the given Drive item. It
@@ -42,34 +63,18 @@ class GCPInterface:
         as defined in :class:`GCPDrivePermission`.
         """
         perms: List[GCPDrivePermission] = []
-        page_count = 0
         fields = [
             "permissions/id",
             "permissions/emailAddress",
         ]
         # See http://googleapis.github.io/google-api-python-client/docs/dyn/drive_v3.permissions.html#list # noqa
         # pylint: disable=no-member
-        next_page_req = self.drive.permissions()\
-            .list(fileId=drive_id,
-                  fields=", ".join(fields),
-                  pageSize=50)
-        while next_page_req is not None:
-            list_res = next_page_req.execute()
-            permissions: List[Any] = list_res['permissions']
-            for p in permissions:
-                if 'emailAddress' in p:
-                    perm = GCPDrivePermission(p['id'], p['emailAddress'])
-                    perms.append(perm)
-
-            # set state for the next page
-            # see https://googleapis.github.io/google-api-python-client/docs/dyn/drive_v3.permissions.html#list_next # noqa
-            # pylint: disable=no-member
-            next_page_req = self.drive.permissions()\
-                .list_next(next_page_req, list_res)
-            page_count += 1
-
-        logging.info(f"Found {len(perms)} permissions across {page_count} "
-                     + f"pages for {drive_id}")
+        for p in self.get_paginated_permissions(drive_id, fields):
+            if 'emailAddress' in p:
+                perm = GCPDrivePermission(p['id'], p['emailAddress'])
+                perms.append(perm)
+
+        logging.info(f"Found {len(perms)} permissions for {drive_id}")
         return perms
 
     def get_parents_permissions(self,

```

Reply:
```
---------------------------------------------------------------- Captured log call -----------------------------------------------------------------
WARNING  root:gcp.py:118 Unable to fetch parents for drive item(team, target-drive): name 'Iterator' is not defined
ERROR    root:gcp.py:141 Failed to load permissions for drive item(team, target-drive): name 'Iterator' is not defined
```

Working on applying a version of your patch

Reply:
1df27bbb12f673fea18436c797373709f8ee2536

I'm against having the paginated function be a to-level API for a similar reason to https://github.com/ubclaunchpad/rocket2/pull/563#discussion_r506576359 (don't expose unused APIs), so I scoped it to the only function that will use it (this one)